File: GameLibrary/src/cc/lib/checkerboard/Suicide.java
package cc.lib.checkerboard;

/**
 * same as checkers only winner is the first to sacrifice all pieces or otherwise fail to be able to move
 */
public class Suicide extends Checkers {
    @Override
    int getWinner(Game game) {
        int winner;
        switch (winner=super.getWinner(game)) {
            case Game.NEAR:
                return Game.FAR;
            case Game.FAR:
                return Game.NEAR;
        }
        return winner;
    }

    @Override
    public long evaluate(Game game, Move move) {
        return -1L * super.evaluate(game, move);
    }

    @Override
    public boolean isJumpsMandatory() {
        return true;
    }

    @Override
    String getDescription() {
        return "Objective: to LOSE all your pieces";
    }
}

File: Checkerboard/src/cc/lib/checkerboard/Suicide.kt
package cc.lib.checkerboard

/**
 * same as checkers only winner is the first to sacrifice all pieces or otherwise fail to be able to move
 */
class Suicide : Checkers() {
	override fun getWinner(game: Game): Int {
		var winner: Int
		when (super.getWinner(game).also { winner = it }) {
			Game.NEAR -> return Game.FAR
			Game.FAR -> return Game.NEAR
		}
		return winner
	}

	override fun isDraw(game: Game): Boolean {
		return super.isDraw(game)
	}

	override fun evaluate(game: Game): Long {
		return -1L * super.evaluate(game)
	}

	override val isJumpsMandatory: Boolean
		get() = true
	override val description: String
		get() = "Objective: to LOSE all your pieces"
}